src/sleep/sleep.c

```diff
@@ -52,41 +52,31 @@ static int write_hibernate_location_info(const HibernateLocation *hibernate_loca
         assert(hibernate_location);
         assert(hibernate_location->swap);
 
+        if (!STR_IN_SET(hibernate_location->swap->type, "partition", "file"))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Invalid swap type for hibernation: %s",
+                                       hibernate_location->swap->type);
+
         resume_str = FORMAT_DEVNUM(hibernate_location->devno);
+        xsprintf(offset_str, "%" PRIu64, hibernate_location->offset);
 
         r = write_string_file("/sys/power/resume", resume_str, WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
hmm, how does this actually work again? isn't echoing the device info into it what triggers the resume? we don't actually want to do that here. how come this ever worked? /me is confused as he realizes he has no idea how this all actually works...

Reply:
The kernel just simply ignores it (but preserves the value) if the written device doesn't contain a hibernation image already I think.

Reply:
that's ugly is fuck though. is there no better way to configure this without triggering this behaviour?

i mean, are we even sure we destroy the hibernation image sig after resuming from it? does the kernel do that on its own?

Reply:
> that's ugly is fuck though. is there no better way to configure this without triggering this behaviour?

Unfortunately I'm not aware of any...

> i mean, are we even sure we destroy the hibernation image sig after resuming from it? does the kernel do that on its own?

Yeah, it does: https://elixir.bootlin.com/linux/latest/source/kernel/power/swap.c#L1538